contentctl/actions/inspect.py

```diff
@@ -179,82 +185,170 @@ def inspectAppCLI(self, config:inspect)-> None:
+            # it sice we will always write AT LEAST the same number of characters
+            # back as we read (due to the addition of whitespace)
             logfile.seek(0)
             json.dump(j, logfile, indent=3, )
-            
+
         reports = j.get("reports", [])
         if len(reports) != 1:
             raise Exception("Expected to find one appinspect report but found 0")
         verbose_errors = []
-        
+
         for group in reports[0].get("groups", []):
-            for check in group.get("checks",[]):
-                if check.get("result","") in status_types:                                    
+            for check in group.get("checks", []):
+                if check.get("result", "") in status_types:
                     verbose_errors.append(f" - {check.get('result','')} [{group.get('name','NONAME')}: {check.get('name', 'NONAME')}]")
         verbose_errors.sort()
-        
+
         summary = j.get("summary", None)
         if summary is None:
             raise Exception("Missing summary from appinspect report")
         msgs = []
         generated_exception = False
         for key in status_types:
-            if summary.get(key,0)>0:
+            if summary.get(key, 0) > 0:
                 msgs.append(f" - {summary.get(key,0)} {key}s")
                 if key in exception_types:
                     generated_exception = True
-        if len(msgs)>0 or len(verbose_errors):
+        if len(msgs) > 0 or len(verbose_errors):
             summary = '\n'.join(msgs)
             details = '\n'.join(verbose_errors)
             summary = f"{summary}\nDetails:\n{details}"
             if generated_exception:
-                raise Exception(f"AppInspect found [{','.join(exception_types)}] that MUST be addressed to pass AppInspect API:\n{summary}")        
+                raise Exception(f"AppInspect found [{','.join(exception_types)}] that MUST be addressed to pass AppInspect API:\n{summary}")
             else:
-                print(f"AppInspect found [{','.join(status_types)}] that MAY cause a failure during AppInspect API:\n{summary}")            
+                print(f"AppInspect found [{','.join(status_types)}] that MAY cause a failure during AppInspect API:\n{summary}")
         else:
             print("AppInspect was successful!")
-                
+
         return
+
+    def check_detection_metadata(self, config: inspect) -> None:
+        """
+        Using a previous build, compare the savedsearches.conf files to detect any issues w/
+        detection metadata.
+        :param config: an inspect config
+        """
+        # TODO (cmcginley): We should be inspect the same artifact we're passing around from the
+        #   build stage ideally
+        # Unpack the savedsearch.conf of each ESCU package
+        current_build_conf = SavedsearchesConf.init_from_package(
+            package_path=config.getPackageFilePath(include_version=False),
```

Comment:
As discussed over slack, in an ideal world we would be inspecting/testing/publishing the same artifact throughout the pipeline, rather than rebuilding at different stages.

Solutions might include:
1. re-merging build and inspect actions
2. renaming inspect as build_and_inspect and use only that for CI/CD
3. Allow us to specify a prior build to inspect via the CLI

This comment is non-blocking, just raising here to give the discussion more visibility

Reply:
#282